### PR TITLE
Fixed basic import.

### PR DESCRIPTION
**Pull request by mikedurbin**
https://jira.duraspace.org/browse/FCREPO-2215

### CODE REVIEW

**Comment by mikedurbin** [comment]
I was wrong, this also fixes: https://jira.duraspace.org/browse/FCREPO-2195

File: src/main/java/org/fcrepo/importexport/importer/Importer.java [hunk 1]
```diff
@@ -67,6 +67,12 @@
     protected FcrepoClient.FcrepoClientBuilder clientBuilder;
 
     /**
+     * The file, within the export directory for the directory mirrors the container
```
Comment (mikedurbin) [inline src/main/java/org/fcrepo/importexport/importer/Importer.java]: Yeah, I'll try again, though this variable will mostly be useful in the future when we can export from one repo/path and import to another.

File: src/main/java/org/fcrepo/importexport/importer/Importer.java [hunk 2]
```diff
@@ -89,49 +99,84 @@ private FcrepoClient client() {
...
     }
 
     private void importDirectory(final File dir) {
-        for (final File f : dir.listFiles()) {
-            if (f.isDirectory()) {
-                importDirectory(f);
-            } else if (f.isFile()) {
-                importFile(f);
-            }
+        for (final File f : dir.listFiles((f) -> {
```
Comment (mikedurbin) [inline src/main/java/org/fcrepo/importexport/importer/Importer.java]: That is exactly what it does.... files before directories.  It looked clearer before checkstyle made me add all those line breaks to the lambdas.  I'll add a comment about the significance which will perhaps make it clearer.

File: src/main/java/org/fcrepo/importexport/importer/Importer.java [hunk 3]
```diff
@@ -190,20 +236,17 @@ private InputStream modelToStream(final Model model) {
         return new ByteArrayInputStream(buf.toByteArray());
     }
 
-    private URI uriForFile(final File f, final File baseDir) throws URISyntaxException {
+    private String uriPathForFile(final File f, final File baseDir) throws URISyntaxException {
         String relative = baseDir.toPath().relativize(f.toPath()).toString();
-        if (relative.startsWith("rest") && config.getResource().toString().endsWith("/rest")) {
```
Comment (mikedurbin) [inline src/main/java/org/fcrepo/importexport/importer/Importer.java]: Yeah, there's a test that prevents us from importing resources that have the rdf:type fedora:RepositoryRoot that I think takes care of this... that said... I had a lot of trouble understanding what this did.

File: src/main/java/org/fcrepo/importexport/importer/Importer.java (hunk 2 above)
Comment (ajs6f) [inline src/main/java/org/fcrepo/importexport/importer/Importer.java]: I thought we reset Checkstyle not to require pointless and confusing line breaks in lambdas?

File: src/main/java/org/fcrepo/importexport/importer/Importer.java [hunk 4]
```diff
@@ -89,49 +99,84 @@ private FcrepoClient client() {
...
+                    return;
+                }
+                final ResIterator binaryResources = model.listResourcesWithProperty(RDF_TYPE, NON_RDF_SOURCE);
+                if (binaryResources.hasNext()) {
+                    destinationUri = new URI(binaryResources.nextResource().getURI());
+                    logger.info("Importing binary {}", sourceRelativePath);
+                    response = importBinary(destinationUri, sanitize(model));
+                } else {
+                    destinationUri = new URI(config.getResource().toString()
+                            + (config.getResource().toString().endsWith("/") ? "" : "/")
```
Comment (mikedurbin) [inline src/main/java/org/fcrepo/importexport/importer/Importer.java]: You're probably right... I always get confused when there's a semantic difference between including a slash at the end of the path and not.  In some posix program (rsync, for instance, I think) it's significant, but perhaps it never is in URL paths.

Comment (mikedurbin) [inline src/main/java/org/fcrepo/importexport/importer/Importer.java]: Oops... that said, it looks like there's explicit logic in the setResource to ensure there's NOT a trailing slash.  I'll just add one without the inline logic.

**Comment by mikedurbin** [comment]
I believe I've responded to all the comments, but it's not as clear to me from the new github comment system.

**Pull request merged**